### DISCUSSION TITLE
Welcome to MineStat Discussions!

### DISCUSSION THREAD

**Discussion by ldilley** [FragLand/minestat#90] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions that you’re wondering about.
  * Share ideas.
  * Provide constructive criticism.
  * Engage with other community members.

  Feel free to comment below to introduce yourself.

**Comment by MackenzieJDriscoll** [comment]
Love the work you guys did on this.  Quick question from my end.  When I incorporate this into my project (C# project), I am getting a good majority of the data back, but my player list is always null, even if someone is on.  I've attached a screenshot.  Is there anything specific that I should be doing on the server side in order for this to work? Maybe a gotchya that I'm not thinking about?
![image](https://user-images.githubusercontent.com/16767969/172721531-a6f080ec-94b2-45b8-99aa-d315d03abc58.PNG)

**Reply by mindsolve** [reply]
Hi @MackenzieJDriscoll, a warm welcome to MineStat!

Indeed there is a small "gotcha" here: the player list is only provided by servers using the `Json` protocol (`SlpProtocol.Json`). In the screenshot you provided the protocol is `Legacy`.
This might be an error in our protocol detection. Could you try and pin the protocol to `Json` by providing the optional parameter `protocol` (if the server you are requesting is newer than MC 1.7)?
Example call would be `MineStat ms = new MineStat("<hostname>", <port>, protocol: SlpProtocol.Json);`

If this doesn't work, or you have further questions, feel free [to open a new discussion](https://github.com/FragLand/minestat/discussions/new) :+1:.

Best regards,
MindSolve

**Comment by LeonBisai** [comment]
Hello, is there a method to get back a list of player names?

**Reply by LeonBisai** [reply]
So in the java library

**Reply by ldilley** [reply]
Hello, @LeonBisai.

No; the Java implementation currently lacks support for the UT3/GS4 protocol.

**Reply by LeonBisai** [reply]
Could you please add that?

**Reply by ldilley** [reply]
It's part of issue #143.
